src/gui/curses/gui-curses-key.c

```diff
@@ -494,13 +494,24 @@ gui_key_flush (int paste)
         /* prepare incomplete UTF-8 char for next iteration */
         if (utf_partial_char[0])
             strcpy (key_str, utf_partial_char);
-        else
+        else if (!paste)
             key_str[0] = '\0';
         length_key_str = strlen (key_str);
 
         /* set last key used in buffer if combo buffer is empty */
         if (gui_key_grab || gui_mouse_event_pending || !gui_key_combo_buffer[0])
             last_key_used = i;
+
+        if (paste && (length_key_str == 63 || i == gui_key_buffer_size - 1)) {
+            gui_buffer_undo_snap (gui_current_window->buffer);
```

Comment:
I'm not sure about the undo stuff, but it seemed like this was how it's done other places.

Reply:
This is changed, so the question is not relevant anymore.